app/uk/gov/hmrc/gform/testonly/SnapshotADTs.scala

```diff
@@ -16,78 +16,166 @@
 
 package uk.gov.hmrc.gform.testonly
 
-import uk.gov.hmrc.mongo.cache.CacheItem
 import java.time.Instant
 import julienrf.json.derived
 import play.api.libs.json.Reads._
 import play.api.libs.json._
+import uk.gov.hmrc.crypto.{ Decrypter, Encrypter }
+import uk.gov.hmrc.gform.save4later.EncryptedFormFormat
+import uk.gov.hmrc.gform.sharedmodel.form.Form
+import uk.gov.hmrc.gform.sharedmodel.form.FormData
+import uk.gov.hmrc.gform.sharedmodel.form.FormId
+import uk.gov.hmrc.gform.sharedmodel.formtemplate.FormTemplateId
+import uk.gov.hmrc.gform.sharedmodel.formtemplate.FormTemplateRaw
 
 case class Snapshot(
```

Comment:
Why we need this type? Why not use `SnapshotItem` instead?

Reply:
`SnapshotItem` also contains a template. I just wanted a ADT that is not bloated with form data and template but can be usefull to  display  a List[Snapshot]  on the restore page. It will be huge if I have to send List[SnapshotItem]